vars/googleCloudFunctionDeploy.groovy

```diff
@@ -103,64 +88,25 @@ def getGcloudPodYaml() {
     """
 }
 
-def copyArchive(String functionName) {
-    container('gcloud') {
-        script {
-            try {
-                def archiveName = "${functionName}.zip"
-                copyArtifacts(
-                    projectName: env.JOB_NAME, 
-                    selector: specific(env.BUILD_NUMBER), 
-                    filter: archiveName, 
-                    fingerprintArtifacts: true
-                )
-                
-                sh """
-                    if [ ! -f "${archiveName}" ]; then
-                        echo "Error: Archive file not found in current directory"
-                        exit 1
-                    fi
-                """
-            } catch (Exception e) {
-                error "Failed to process function archive: ${e.message}"
-            }
-        }
-    }
-}
-
-def uploadArchive(Map config) {
-    sh """
-        gcloud config set project ${config.projectId}
-        gcloud storage cp ${config.functionName}.zip ${config.gcsPath}
-    """
-}
-
 def deployFunction(Map config) {
-    def deployCommand = "gcloud functions deploy ${config.functionName}"
-    
-    if (config.generation == 'gen2') {
-        // Jenkins is very sensitive to indentation, so this snippet looks strange but it's correct
-        deployCommand += """ --gen2 \\
-        --concurrency ${config.concurrency ? config.concurrency : 1} \\
-        --cpu ${config.cpu ? config.cpu : 1} \\"""
-    } else {
-        // Have to explicitly set since gen2 is becoming the default
-        deployCommand += " --no-gen2 \\"
-    }
-    
+    def deployCommand = "gcloud functions deploy ${config.functionName} \\"
+
     deployCommand += """
+        --gen2 \\
+        --project ${config.projectId} \\
         --runtime ${config.runtime} \\
         --region ${config.region} \\
-        --source ${config.gcsPath} \\
-        --service-account ${config.serviceAccountEmail} \\
-        --set-env-vars ${config.environmentVariables.collect { "${it.key}=${it.value}" }.join(',')} \\
```

Comment:
How would it know which service account to use?

Reply:
Thank you. Missed that. Updating and testing

Reply:
I've just confirmed a fix for this